### PR TITLE
Connect MoviePage with api and MoviesListPage

### PR DESCRIPTION
**Pull request by greedyChicken**
done:
- displaying MoviePage data from api call, 
- redirecting to MoviePage from MoviesListPage items,
- tidied catalog structure for MoviePage,

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -8,15 +8,15 @@ const App = () => (
   <HashRouter>
     <Header />
     <Switch>
+      <Route path={"/movies/:id"}>
+        <MoviePage />
+      </Route>
```
Comment (bartekdbc) [inline src/App.js:13]: That was necessary change. Now the app is more complex, great!

File: src/features/movies/MoviePage/index.js [hunk 2]
```diff
@@ -1,49 +1,101 @@
...
+        <Loader />
+      ) : error || !movie?.id ? (
+        <ErrorPage />
+      ) : (
+        <>
+          <BigMovieTile
+            poster={
+              movie.backdrop_path &&
+              `${APIImageUrl}/original${movie.backdrop_path}`
+            }
```
Comment (bartekdbc) [inline src/features/movies/MoviePage/index.js:48]: Just to remind you in this place - movie title, voteAverage (with star icon) and voteCount need to be entered on the backdrop poster.

File: src/common/TileLink/index.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { Link } from "react-router-dom";
+import styled from "styled-components";
+
+export const TileLink = styled(Link)`
+  text-decoration: none;
+`;
```
Comment (bartekdbc) [inline src/common/TileLink/index.js:6]: TileLink prepared for later needs of making PersonPage - awesome.

File: src/features/movies/MoviesListPage/moviesSaga.js [hunk 4]
```diff
@@ -9,7 +9,7 @@ import {
 
 function* fetchPopularMoviesHandler() {
   try {
-    yield delay(1000);
+    yield delay(200);
```
Comment (bartekdbc) [inline src/features/movies/MoviesListPage/moviesSaga.js:12]: Good change - 200 for now is enough.

File: src/common/PopularMoviesTile/index.js [hunk 5]
```diff
@@ -40,7 +39,7 @@ export const PopularMoviesTile = ({
         <Release>{date}</Release>
         <Tags>
           {genres.map((genre) => (
-            <TagButton key={nanoid()}>
+            <TagButton key={genre.id}>
```
Comment (bartekdbc) [inline src/common/PopularMoviesTile/index.js:42]: As we agreed, we change the nanoid to the id of a specific category - great!

File: src/features/people/PeopleListPage/index.js [hunk 6]
```diff
@@ -42,7 +41,7 @@ const PeopleListPage = () => {
             <Layout>
               {popularPeople?.map((person) => (
                 <PersonTile
-                  key={nanoid()}
+                  key={person.id}
```
Comment (bartekdbc) [inline src/features/people/PeopleListPage/index.js:44]: As we agreed, we change the nanoid to the id of a specific category - great!

File: src/features/utilities.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    }
+  );
+
+  // getting exact number of people
+  filteredArray = filteredArray.filter(
+    (item) => filteredArray.indexOf(item) < numberOfItems
+  );
+
+  return filteredArray;
+};
```
Comment (bartekdbc) [inline src/features/utilities.js:35]: It's good that you pulled such functions into a separate file, now the code is very clear. Formatting the date and changing the presentation of people is a really good idea!

File: src/common/PersonTile/index.js [hunk 8]
```diff
@@ -20,7 +20,7 @@ export const PersonTile = ({ profile, profilePath, fullName }) => {
       )}
       <PersonDetails>
         <PersonName>{fullName}</PersonName>
-        <PersonFunction>{""}</PersonFunction>
+        <PersonFunction>{role}</PersonFunction>
```
Comment (bartekdbc) [inline src/common/PersonTile/index.js:23]: Good that you noticed that this part of the code needed a change.

File: src/features/movies/MoviePage/movieSaga.js [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+    yield put(fetchMovieSuccess(movie));
+    yield put(fetchMovieCreditsSuccess(movieCredits));
+  } catch (error) {
+    yield put(fetchMovieError());
+  }
+}
+
+export function* watchFetchMovie() {
+  yield takeLatest(fetchMovieId.type, fetchMovieHandler);
+}
```
Comment (bartekdbc) [inline src/features/movies/MoviePage/movieSaga.js:24]: I think it's very nice that you tidied up the file structure - separate Saga file for single Movie, and separately for the MovieList.

File: src/features/movies/MoviePage/movieSlice.js [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+const selectMovieState = (state) => state.movie;
+
+export const selectLoading = (state) => selectMovieState(state).loading;
+export const selectError = (state) => selectMovieState(state).error;
+export const selectMovie = (state) => selectMovieState(state).movie;
+export const selectMovieId = (state) => selectMovieState(state).movieId;
+export const selectMovieCredits = (state) =>
+  selectMovieState(state).movieCredits;
+
+export const movieReducer = movieSlice.reducer;
```
Comment (bartekdbc) [inline src/features/movies/MoviePage/movieSlice.js:50]: Separate Slice file for single Movie, and separate for the MovieList - it was something that needed to be done!

**Review by bartekdbc — APPROVED** [review]
Overall you did really GREAT JOB! You have introduced a lot of new interactions into app and made a lot of necessary changes in code -  all I can say is that the end result is really impressive. 

One thing to note - after implementing all the logic, in next step we also need to take care of the responsiveness of our website, because I noticed many shortcomings in this matter.

File: src/App.js (hunk 1 above)
Comment (GrzegorzPacewicz) [reply src/App.js:13]: I cant wait merge. It looks like it's a big step.

**Review by GrzegorzPacewicz — COMMENTED** [review]
I cant wait merge. It's a big step toward make this app done

**Pull request merged**